scripts/utils/environment.js

```diff
@@ -9,3 +12,76 @@ export const execOptions = { encoding: "utf8", stdio: "inherit" };
 export function configurePython(options = execOptions) {
   execSync("pipenv --python 3.8.17", options);
 }
+
+export function preExecuteChecks(pythonCheck = false, cookiecutterCheck = false) {
+  // Check if Node.js v18.16.0 is installed
+  try {
+    section("Checking Node version");
+    const nodeVersion = execSync("node --version", {
+      cwd: userdir,
+      shell: true,
+      encoding: "utf8"
+    }).trim();
+    if (!nodeVersion.includes("v18.16.0")) {
+      section(`Node.js v18.16.0 is not installed. Found version: ${nodeVersion}. Please install Node.js v18.16.0 before running this script.`);
+      process.exit(1);
+    }
+  } catch (error) {
+    section("Node.js v18.16.0 is not installed. Please install Node.js v18.16.0 before running this script.");
+    process.exit(1);
+  }
+
+  // Check if Python 3.8.17 is installed
+  if (pythonCheck) {
+    try {
+      section("Checking Python version");
+      // Fallback is for older versions of Python and is why we used spawnSync. We can use execSync if this distinction really doesn't matter.
+      const pythonVersion = spawnSync("python", ["--version"], {
+        cwd: userdir,
+        shell: true,
+        encoding: "utf8"
+      }).stdout || spawnSync("python", ["--version"], { shell: true, encoding: "utf8" }).stderr;
+      if (!pythonVersion.includes("3.8")) {
+        section(`Python 3.8 is not installed. Found version: ${pythonVersion}. Please install and try again.`);
+        process.exit(1);
```

Comment:
This check should be a regular expression so that if the user has Python 3.8.2 for example, it won't fail. it's still within the main version range. But doing so using the python-version file